include/SDL3/SDL_gpu.h

```diff
@@ -0,0 +1,2263 @@
+    SDL_GPU_BACKEND_VULKAN = 0x0000000000000001,
+    SDL_GPU_BACKEND_D3D11 = 0x0000000000000002,
+    SDL_GPU_BACKEND_METAL = 0x0000000000000004,
+    SDL_GPU_BACKEND_ALL = (SDL_GPU_BACKEND_VULKAN | SDL_GPU_BACKEND_D3D11 | SDL_GPU_BACKEND_METAL)
+} SDL_GpuBackendBits;
+
+typedef Uint64 SDL_GpuBackend;
+
+/* Structures */
+
+typedef struct SDL_GpuDepthStencilValue
+{
+    float depth;
+    Uint32 stencil;
+} SDL_GpuDepthStencilValue;
+
+typedef struct SDL_GpuRect
+{
+    Sint32 x;
+    Sint32 y;
+    Sint32 w;
+    Sint32 h;
+} SDL_GpuRect;
+
+typedef struct SDL_GpuColor
+{
+    float r;
+    float g;
+    float b;
+    float a;
+} SDL_GpuColor;
+
+typedef struct SDL_GpuViewport
+{
+    float x;
+    float y;
+    float w;
+    float h;
+    float minDepth;
+    float maxDepth;
+} SDL_GpuViewport;
+
+typedef struct SDL_GpuTextureTransferInfo
+{
+    SDL_GpuTransferBuffer *transferBuffer;
+    Uint32 offset;      /* starting location of the image data */
+    Uint32 imagePitch;  /* number of pixels from one row to the next */
+    Uint32 imageHeight; /* number of rows from one layer/depth-slice to the next */
+} SDL_GpuTextureTransferInfo;
+
+typedef struct SDL_GpuTransferBufferLocation
+{
+    SDL_GpuTransferBuffer *transferBuffer;
+    Uint32 offset;
+} SDL_GpuTransferBufferLocation;
+
+typedef struct SDL_GpuTransferBufferRegion
+{
+    SDL_GpuTransferBuffer *transferBuffer;
+    Uint32 offset;
+    Uint32 size;
+} SDL_GpuTransferBufferRegion;
+
+typedef struct SDL_GpuTextureSlice
+{
+    SDL_GpuTexture *texture;
+    Uint32 mipLevel;
+    Uint32 layer;
+} SDL_GpuTextureSlice;
+
+typedef struct SDL_GpuTextureLocation
+{
+    SDL_GpuTextureSlice textureSlice;
+    Uint32 x;
+    Uint32 y;
+    Uint32 z;
+} SDL_GpuTextureLocation;
+
+typedef struct SDL_GpuTextureRegion
+{
+    SDL_GpuTextureSlice textureSlice;
+    Uint32 x;
+    Uint32 y;
+    Uint32 z;
+    Uint32 w;
+    Uint32 h;
+    Uint32 d;
+} SDL_GpuTextureRegion;
+
+typedef struct SDL_GpuBufferLocation
+{
+    SDL_GpuBuffer *buffer;
+    Uint32 offset;
+} SDL_GpuBufferLocation;
+
+typedef struct SDL_GpuBufferRegion
+{
+    SDL_GpuBuffer *buffer;
+    Uint32 offset;
+    Uint32 size;
+} SDL_GpuBufferRegion;
+
+typedef struct SDL_GpuIndirectDrawCommand
+{
+    Uint32 vertexCount;   /* number of vertices to draw */
+    Uint32 instanceCount; /* number of instances to draw */
+    Uint32 firstVertex;   /* index of the first vertex to draw */
+    Uint32 firstInstance; /* ID of the first instance to draw */
+} SDL_GpuIndirectDrawCommand;
+
+typedef struct SDL_GpuIndexedIndirectDrawCommand
+{
+    Uint32 indexCount;    /* number of vertices to draw */
+    Uint32 instanceCount; /* number of instances to draw */
+    Uint32 firstIndex;    /* base index within the index buffer */
+    Uint32 vertexOffset;  /* value added to vertex index before indexing into the vertex buffer */
+    Uint32 firstInstance; /* ID of the first instance to draw */
+} SDL_GpuIndexedIndirectDrawCommand;
```

Comment:
Any plans for indirect compute dispatches?

Reply:
Tracking: https://github.com/thatcosmonaut/SDL/issues/107